src/libcore/iter/iterator.rs

```diff
@@ -2463,6 +2463,52 @@ pub trait Iterator {
             }
         }
     }
+
+    /// Extends passed collection with the contents of an iterator.
+    ///
+    /// This method consumes the iterator and adds all its items to the
+    /// passed collection. The collection is then returned, so the call chain
+    /// can be continued.
+    ///
+    /// This method is a counter-part of `Extend::extend`, but instead of
```

Comment:
You should mention this in the documentation of `Extend` with a note that the methods `Iterator::extend(_mut)` are preferred.